### ISSUE TITLE
Create `ProxyKind` specifically for bidding on parathread blockspace

### ISSUE THREAD

**Issue by eskimor** [paritytech/polkadot#6733]
The parathread bidding use-case requires an automated account on a collator to be funded and controlled by node-side logic, which will determine when to send a bid, how much, etc. In order to limit attack surface, it would be best for collators hold hot signing keys which only have the ability to bid on parathread blockspace and not transfer/governance/etc. All runtimes where parathread bidding is deployed should have a custom `ProxyKind` which is only allowed to issue parathread bids.

**Comment by eskimor** [comment]
Hint by Rob:

Proxies are quite simple to implement, actually. There is a ProxyKind enum defined in every runtime that uses the proxy pallet, and it implements CallFilter or some similar trait that allows allowed call types by the proxy to be filtered.

**Comment by kianenigma** [comment]
Yeah adding this is super trivial. @Ank4n recently did a PR on this, might be able to chime in here too.
